### DISCUSSION TITLE
How to implement multiple layouts?

### DISCUSSION THREAD

**Discussion by oburlaca** [vuetifyjs/vuetify#12726] [Q&A]
I'm building a SPA, it has a Login page (just a login button) and several APP pages that has AppBar, BottomNav. 
What the best way to have multiple layouts in a SPA? 

From the [documentation](https://next.vuetifyjs.com/en/components/application/):
> v-app should only be rendered within your application **ONCE**

But then it says:
> If you are using multiple layouts in your application you will need to ensure **each root layout file** that will contain Vuetify components has a **v-app at the root** of its template.

I'm a bit confused: I need to include v-app in each layout file, but at the same time it should be rendered ONCE.

Are there examples how to build a multiple layout SPA using Vuetify? 
The docs about [Layouts](https://next.vuetifyjs.com/en/features/layouts/) talks about single layout app.

**Comment by AGDholo** [comment]
My understanding is that different layouts are used to route the root, and each layout should have v-app.

Example:

My site have the landing page and dashboard, so it need two layout.

https://github.com/AGDholo/shock/tree/master/src/components/root
https://github.com/AGDholo/shock/blob/master/src/router/index.js

**Reply by oburlaca** [reply]
Is it ok to have `<v-app>` inside `<v-app>` ?
Your [App.vue](https://github.com/AGDholo/shock/blob/master/src/App.vue) file has `<v-app>`, and [front.vue](https://github.com/AGDholo/shock/blob/master/src/components/root/front.vue) also has a `<v-app>`, while [dashboard.vue](https://github.com/AGDholo/shock/blob/master/src/components/root/dashboard.vue) doesn't.

**Comment by oburlaca** [comment]
@AGDholo, thanks for your example, it's similar to the solutions I found.
For ex this one: https://dev.to/lampewebdev/vuejs-pages-with-dynamic-layouts-problems-and-a-solution-4460

But I was thinking I'm doing something wrong, because switching layouts will cause some animation in `<v-main>`.
I found the issue appears when using `<keep-alive>`.

Here is a small demo that illustrates the effect: https://github.com/oburlaca/layout-test
 
Notice the keep-alive in [App.vue](https://github.com/oburlaca/layout-test/blob/main/src/App.vue).
When I switch from [MainAppLayout](https://github.com/oburlaca/layout-test/blob/main/src/layouts/MainAppLayout.vue) to [SimpleLayout](https://github.com/oburlaca/layout-test/blob/main/src/layouts/SimpleLayout.vue) and back, there is an animation/transition.

Here is a screencast, notice the scrolling of the list when the layout is switched:

![layout-test](https://user-images.githubusercontent.com/1470654/101804360-2cb24080-3b1a-11eb-94f7-14aca8512905.gif)

If I remove **`<keep-alive>`** from App.vue, the animation disappears, but the components are recreated each time.

So I narrow my initial question to "How to get rid of the animation when switching layouts?".
